pyuvdata/tests/test_uvdata.py

```diff
@@ -1102,6 +1102,54 @@ def test_select_times():
     pytest.raises(ValueError, uv_object.select, times=[np.min(unique_times) - uv_object.integration_time[0]])
 
 
+@pytest.mark.filterwarnings("ignore:Telescope EVLA is not")
+def test_select_time_range():
+    uv_object = UVData()
+    testfile = os.path.join(
+        DATA_PATH, 'day2_TDEM0003_10s_norx_1src_1spw.uvfits')
+    uv_object.read_uvfits(testfile)
+    old_history = uv_object.history
+    unique_times = np.unique(uv_object.time_array)
+    mean_time = np.mean(unique_times)
+    time_range = [np.min(unique_times), mean_time]
+    times_to_keep = unique_times[np.nonzero((unique_times <= time_range[1])
+                                            & (unique_times >= time_range[0]))]
+
+    Nblts_selected = np.nonzero((uv_object.time_array <= time_range[1])
+                                & (uv_object.time_array >= time_range[0]))[0].size
+
+    uv_object2 = copy.deepcopy(uv_object)
+    uv_object2.select(time_range=time_range)
+
+    assert times_to_keep.size == uv_object2.Ntimes
+    assert Nblts_selected == uv_object2.Nblts
+    for t in times_to_keep:
+        assert t in uv_object2.time_array
+    for t in np.unique(uv_object2.time_array):
+        assert t in times_to_keep
+
+    assert uvutils._check_histories(old_history + '  Downselected to '
+                                    'specific times using pyuvdata.',
+                                    uv_object2.history)
+
+    # check for error associated with times not included in data
+    with pytest.raises(ValueError) as cm:
+        uv_object.select(time_range=[np.min(unique_times) - uv_object.integration_time[0] * 2,
+                                     np.min(unique_times) - uv_object.integration_time[0]])
+    assert str(cm.value).startswith('No elements in time range')
+
+    # check for error setting times and time_range
+    times_to_keep = unique_times[[0, 3, 5, 6, 7, 10, 14]]
+    with pytest.raises(ValueError) as cm:
+        uv_object.select(time_range=time_range, times=times_to_keep)
+    assert str(cm.value).startswith('Only one of "times" and "time_range" can be set')
+
+    # check for errors time_range not length 2
+    with pytest.raises(ValueError) as cm:
+        uv_object.select(time_range=time_range[0])
+    assert str(cm.value).startswith('time_range must be length 2')
```

Comment:
I know they use the same setup as the initial selection test, but can we please break these error checks into separate tests.

Reply:
sure